chia/consensus/blockchain.py

```diff
@@ -809,23 +803,20 @@ async def get_block_records_at(self, heights: List[uint32], batch_size=900) -> L
         gets block records by height (only blocks that are part of the chain)
         """
         records: List[BlockRecord] = []
-        hashes = []
+        hashes: List[bytes32] = []
         assert batch_size < 999  # sqlite in python 3.7 has a limit on 999 variables in queries
         for height in heights:
-            hashes.append(self.height_to_hash(height))
+            header_hash: Optional[bytes32] = self.height_to_hash(height)
+            if header_hash is None:
+                raise ValueError(f"Do not have block at height {height}")
```

Comment:
Would this have resulted in an exception before? Or just not gotten a record from the db?  I looked down to the query but haven't traced back up the call stack to see the effect of an incomplete result.

Reply:
Probably an exception in DB, which would  make the whole API call or weight proof creation process fail

Reply:
My comment was that the db looked like it would just not match anything thus continuing on instead of raising an exception.